### PR TITLE
Default to HTTP/2 on dotnet5

### PR DESCRIPTION
**Pull request by brendandburns**

### CODE REVIEW

**Comment by tg123** [comment]
let me cherrypick and test
seems i missed `DefaultRequestVersion = 2` last time

**Comment by brendandburns** [comment]
yeah, there's no version negotiation in HTTP :)

File: src/KubernetesClient/Kubernetes.ConfigInit.cs [hunk 1]
```diff
@@ -243,6 +246,10 @@ private void CreateHttpClient(DelegatingHandler[] handlers, KubernetesClientConf
 
             AppendDelegatingHandler<WatcherDelegatingHandler>();
             HttpClient = new HttpClient(FirstMessageHandler, false);
+
+#if NET5_0
+            HttpClient.DefaultRequestVersion = HttpVersion.Version20;
```
Comment (tg123) [inline src/KubernetesClient/Kubernetes.ConfigInit.cs:251]: This does NOT work, 
see <https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpclient.defaultrequestversion?view=net-5.0>

> The DefaultRequestVersion property doesn't apply to the **SendAsync** method. The HttpRequestMessage parameter passed as the argument to the SendAsync method has its own Version property that controls the HTTP version used for the request.

Unfortunately, our generated code use `SendAsync` LOL


Here is my test (Should find a way to do automation)

1. create a AKS (1.19)
1. disable tcp keepalive
1. start watch example

got a [R] from remote side after 5 min and then `watch` will not show any event from `kubectl run` 

```
02:42:58.973029 IP 10.x.x.x.49746 > 52.x.x.x.443: Flags [.], ack 28897, win 492, options [nop,nop,TS val 255369207 ecr 3243887534], length 0
02:47:48.004159 IP 52.x.x.x.443 > 10.x.x.x.49746: Flags [R], seq 660228079, win 0, length 0
```

**Comment by tg123** [comment]
please pick #592 into this PR
and I tested requests to `http` will down to http1.1 automatically

**Comment by alvaroaleman** [comment]
@brendandburns: PR needs rebase.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Pull request closed without merging**